include/deal.II/base/mpi.templates.h

```diff
@@ -119,6 +120,28 @@ namespace Utilities
 #ifdef DEAL_II_WITH_MPI
         if (job_supports_mpi())
           {
+#ifdef DEBUG
+            {
+              const unsigned int rank = this_mpi_process(mpi_communicator);
+              int size = values.size();
```

Comment:
You can write this whole block a lot simpler if you do just this:
```
  Assert (Utilities::MPI::min (values.size(), mpi_communicator) == values.size(), ExcMessage(...));
  Assert (values.size() == output.size(), ExcMessage(...));
```
This way, if one of the processes has the wrong input or output size, then you will get an exception on at least one process. It's not guaranteed (like in your code) that *all* processes abort. But in practice, it is enough if *one* of them does. The assertions above trigger on at least one process unless *all processes have the same size*.

Reply:
> You can write this whole block

nope, i can't use `Utilities::MPI::min` because it will call this function recursively!
I also think `min` is not enough, you should at least call `max` among all of them, but i can probably drop the same checks for `values`.

Reply:
> EDIT: min indeed should be enough. But still, need to do manually.

actually, it depends. I currently do `MPI_Reduce`, not `allreduce`, so in this case I need both min/max to be sure.